File: src/backend/traintickets/business-logic/src/main/java/traintickets/businesslogic/model/Ticket.java
package traintickets.businesslogic.model;

import traintickets.businesslogic.exception.InvalidEntityException;

import java.io.Serializable;
import java.math.BigDecimal;

public record Ticket(TicketId ticketId,
                     UserId owner,
                     String passenger,
                     RaceId race,
                     int railcar,
                     Place place,
                     Schedule start,
                     Schedule end,
                     BigDecimal cost) implements Serializable {
    public void validate() {
        if (owner == null || passenger == null || race == null || railcar < 1 ||
                place == null || start == null || end == null || cost == null) {
            throw new InvalidEntityException("All data required");
        }
        place.validate();
        start.validate();
        end.validate();
        if (cost.compareTo(BigDecimal.ZERO) < 0) {
            throw new InvalidEntityException("Invalid cost");
        }
    }
}

File: src/frontend/tech-ui/src/main/kotlin/model/Ticket.kt
package traintickets.console.model

import kotlinx.serialization.Serializable

@Serializable
data class Ticket(
    val ticketId: TicketId? = null,
    val owner: UserId,
    val passenger: String,
    val race: RaceId,
    val railcar: Int,
    val place: Place,
    val start: Schedule,
    val end: Schedule,
    val cost: String,
)